SignalServiceKit/src/Profiles/OWSProfilesManager.m

```diff
@@ -375,32 +436,164 @@ - (void)updateProfileOnService:(nullable NSString *)localProfileName
     });
 }
 
-- (void)loadLocalProfileAsync
+#pragma mark - Profile Whitelist
+
+- (void)addUserToProfileWhitelist:(NSString *)recipientId
+{
+    OWSAssert(recipientId.length > 0);
+
+    [self.dbConnection setObject:@(1) forKey:recipientId inCollection:kOWSProfilesManager_UserWhitelistCollection];
+    self.userProfileWhitelistCache[recipientId] = @(YES);
```

Comment:
nit: Would it be simpler to have `userProfileWhitelistCache` be an NSSet of recipientIds, and maybe use `- (BOOL)hasObjectForKey:(NSString *)key inCollection:(nullable NSString *)collection;` rather than worrying about the value?

edit: I guess the current implementation makes it's easier to build a lazily populated cache. Makes sense as is!